### PR TITLE
Use pre-commit for testing style

### PR DESCRIPTION
**Pull request by oyvindeide**
After the addition of `ruff`, the style check is really fast, and I have personally used this configuration for a while now, which removes the problem of committing something, only to notice `style` is failing.

This is of course opt-in, which means no one is forced to use it locally, it is fully voluntary.

(Screenshot of new behavior in GUI if applicable)

## Pre review checklist

- [x] Read through the code changes carefully after finishing work
- [x] Make sure tests pass locally (after every commit!)
- [x] Prepare changes in small commits for more convenient review (optional)
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Updated documentation
- [x] Ensured that unit tests are added for all new behavior (See 
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    -   id: check-json
+    -   id: check-yaml
+    -   id: end-of-file-fixer
+    -   id: trailing-whitespace
+
+-   repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.1.6
+    hooks:
+    - id: ruff
+      args: [ --fix ]
```
Comment (berland) [inline .pre-commit-config.yaml:14]: In this PR, it seems that pre-commit is also used on the Github Action side, won't this ` --fix` then make code with fixable ruff-errors pass CI?

Comment (oyvindeide) [reply .pre-commit-config.yaml:14]: Good point! There is apparently `--exit-non-zero-on-fix`, will add that!

Comment (oyvindeide) [reply .pre-commit-config.yaml:14]: Scratch that, seems to work as expected for the hook:
```
ruff.....................................................................Failed
- hook id: ruff
- files were modified by this hook

Found 1 error (1 fixed, 0 remaining).
```

**Review by sondreso — APPROVED** [review]
:art:

**Pull request merged**